pkg/policy/l4.go

```diff
@@ -35,6 +35,29 @@ const (
 	ParserTypeKafka L7ParserType = "kafka"
 )
 
+func httpEqual(a, b api.PortRuleHTTP) bool {
+	if a.Path != b.Path ||
+		a.Method != b.Method ||
+		a.Host != b.Host ||
+		len(a.Headers) != len(b.Headers) {
+		return false
+	}
+
+	for i, value := range a.Headers {
+		if b.Headers[i] != value {
+			return false
+		}
+	}
+	return true
+}
+
+func kafkaEqual(a, b api.PortRuleKafka) bool {
+	if a.APIVersion != b.APIVersion || a.APIKey != b.APIKey || a.Topic != b.Topic {
```

Comment:
Why not `return a.APIVersion == b.APIVersion && a.APIKey == b.APIKey && a.Topic == b.Topic`?

Reply:
No reason other that copy paste from HTTP case that required looping through the header matches after. I changed this as you suggested :-)